File: app/models/question.rb
class Question < ActiveRecord::Base
  belongs_to :question_round

  has_many :answers, -> { order('id ASC') }
  accepts_nested_attributes_for :answers, allow_destroy: true

  has_many :interactive_sources, as: :subject
  accepts_nested_attributes_for :interactive_sources, allow_destroy: true

  has_many :image_sources, as: :subject
  accepts_nested_attributes_for :image_sources, allow_destroy: true

  %i(title body correct_title incorrect_title explanation_source).each do |translated_field|
    translates translated_field
  end
end

File: app/assets/javascripts/models/question.js
Course.Question = DS.Model.extend({
  title: DS.attr('string'),
  explanation_source: DS.attr('string'),
  test_case: DS.attr('boolean'),
  answers: DS.hasMany('answer'),
  interactive_sources: DS.hasMany('interactive_source'),
  image_sources: DS.hasMany('image_source'),
  answered: false,
  correct: false,
  active: false,
  timeAnswered: 0,
  hasInteractives: Ember.computed.notEmpty('interactive_sources'),
  hasInteractiveTriplet: Ember.computed.equal('interactive_sources.length', 3),
  hasImages: Ember.computed.notEmpty('image_sources'),
  incorrect_title: DS.attr('string'),
  correct_title: DS.attr('string')
});
